conkit/plot/contactplotplus.py

```diff
@@ -0,0 +1,267 @@
+            xy_c_aa_name = [self.conpred.sequence.seq[res_idx - 1] for res_idx in xy_c]
+            xy_e_aa_name = [self.conpred.sequence.seq[res_idx - 1] for res_idx in xy_e]
+
+            # add aa properties to each residue from aa property dictionary
+            h_aa_properies = [self.aa_properites[x] for x in xy_h_aa_name]
+            c_aa_properies = [self.aa_properites[x] for x in xy_c_aa_name]
+            e_aa_properies = [self.aa_properites[x] for x in xy_e_aa_name]
+
+            # plot ss on to the contact plot with residue properties
+            source2 = ColumnDataSource(data=dict(x=xy_h, y=xy_h, aa_name=xy_h_aa_name, aa_properties=h_aa_properies))
+            source3 = ColumnDataSource(data=dict(x=xy_c, y=xy_c, aa_name=xy_c_aa_name, aa_properties=c_aa_properies))
+            source4 = ColumnDataSource(data=dict(x=xy_e, y=xy_e, aa_name=xy_e_aa_name, aa_properties=e_aa_properies))
+
+            self.sspred_h = self.canvas.circle(x='x', y='y', size=3, color='orange', source=source2)
+            self.canvas.add_tools(
+                HoverTool(renderers=[self.sspred_h], tooltips=[('Position', '@x'), ('Residue Name', '@aa_name'),
+                                                               ('Properties', '@aa_properties')]))
+            self.sspred_c = self.canvas.circle(x='x', y='y', color='blue', size=3, source=source3)
+            self.canvas.add_tools(
+                HoverTool(renderers=[self.sspred_c], tooltips=[('Position', '@x'), ('Residue Name', '@aa_name'),
+                                                               ('Properties', '@aa_properties')]))
+            self.sspred_e = self.canvas.circle(x='x', y='y', color='pink', size=3, source=source4)
+            self.canvas.add_tools(
+                HoverTool(renderers=[self.sspred_e], tooltips=[('Position', '@x'), ('Residue Name', '@aa_name'),
+                                                               ('Properties', '@aa_properties')]))
+        else:
+            raise ValueError('ss prediction is not the same length as sequence')
+
+    def add_dispred_layer(self, dispred):
+
+        self.dispred = dispred
+
+        if len(dispred) == len(self.conpred.sequence.seq):
+
+            x_d = [residue.id for residue in self.dispred.search(0.5)]
+            x_nd = [residue.id for residue in self.dispred if residue.id not in x_d]
+            y_d = [residue.id for residue in self.dispred.search(0.5)]
+            y_nd = [residue.id for residue in self.dispred if residue.id not in y_d]
+            y_d = [int(y) - 3 for y in y_d]
+            y_nd = [int(y) - 3 for y in y_nd]
+
+            self.dispred_o = self.canvas.circle(x_nd, y_nd, color='grey', size=5)
+            self.dispred_dis = self.canvas.circle(x_d, y_d, size=5, color='lime')
+        else:
+            raise ValueError('disorder prediction is not the same length as sequence')
+
+    def add_conservpred_layer(self, conservpred):
+        self.conservpred = conservpred
```

Comment:
what happens if I call this (and any other ``add_``) function three times? I add three layers, I set the ``conservpred`` file three times, and this is a terrible idea. why assign this to the instance? keep it local, add the layer, done